### PR TITLE
Simple grid and subtemplates

### PR DESCRIPTION
**Pull request by gdey**
implements simple grid and subtemplates

### CODE REVIEW

File: atlante/README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+Atlante allow one to created printed maps based on templates
```
Comment (ARolek) [inline atlante/README.md]: created -> create

File: atlante/README.md [hunk 2]
```diff
@@ -74,4 +76,14 @@ returns
...
+## Support for sub-templates
+
+If the template file is local (e.g. on the filesystem), then the `templates` directory at the
+same level will be used to load base templates; files who's extension is '.tpl'.
+
+For example: 
+
+Say that the system is loading `templates/50k.svg`, and there is a `templates/templates` directory. Then every file in the `templates/templates` directory with an extension of `.tpl` (e.g. `templates/templates/*.tpl`) will be loading before `templates/50k.svg` is parsed.
+
+This allows one to define callable templates in those files, and override them as need in `templates/50k.svg`
```
Comment (ARolek) [inline atlante/README.md]: need -> needed

File: atlante/sheet.go [hunk 3]
```diff
@@ -62,6 +65,45 @@ type Sheet struct {
 	UseCached bool
 }
 
+func loadTemplateDir(t *template.Template, location *url.URL) (*template.Template, error) {
```
Comment (ARolek) [inline atlante/sheet.go]: please document that this function specifically looks for the directory "templates" (I saw this in the README).

File: atlante/template/grating/grating.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	}
+	rowOffset := (float64(row) * grate.Height)
+	if grate.FlipY {
+		rowOffset *= -1
+	}
+	colOffset := (float64(col) * grate.Width)
+	return geom.Point{
+		grate.Extent[0] + colOffset,
+		grate.Extent[1] + rowOffset,
+	}
```
Comment (ARolek) [inline atlante/template/grating/grating.go]: Looks like the initial return is causing the rest of this code to never be executed.

File: atlante/template_helpers.go [hunk 5]
```diff
@@ -5,42 +5,83 @@ import (
...
+	"rounder3":           tplRound3,
+	"first":              tplFirstNonZero,
+	"DrawBars":           TplDrawBars,
+	"asIntSlice":         IntSlice,
+	"pixel_bounds":       PixelBounds,
+	"join":               tplStringJoin,
+	"split":              tplStringSplit,
+	"idx":                tplIndexOf,
+	"point":              tplNewPoint,
+	"SimpleGridFromArgs": simpleGridFromArgs,
```
Comment (ARolek) [inline atlante/template_helpers.go]: any reason for using a different naming convention for `SimpleGridFromArgs` from the other functions (with the exception of `DrawBars`)?

File: atlante/template_helpers.go [hunk 6]
```diff
@@ -821,3 +815,171 @@ func TplDrawBars(bottomLeft, topRight coord.LngLat, pxlBox PixelBox, grid trelli
...
+		return nil, err
+	}
+	width, err := args.GetAsFloat64("Width")
+	if err != nil {
+		return nil, err
+	}
+	height, err := args.GetAsFloat64("Height")
+	if err != nil {
+		return nil, err
+	}
```
Comment (ARolek) [inline atlante/template_helpers.go]: Maybe move the keys here to `const`s

File: startup.sh [hunk 7]
```diff
@@ -0,0 +1 @@
+docker run --rm -v /Users/gdey/go/src/github.com/go-spatial/maptoolkit:/go/src/github.com/go-spatial/maptoolkit -it atlante-dev /bin/bash
```
Comment (ARolek) [inline startup.sh]: did you mean to add this file?

Comment (gdey) [reply startup.sh]: Nope this is just a helper file for me.

File: atlante/template/grating/grating.go (hunk 4 above)
Comment (gdey) [reply atlante/template/grating/grating.go]: Yup, forgot to remove the old code.

File: atlante/template_helpers.go (hunk 5 above)
Comment (gdey) [reply atlante/template_helpers.go]: I've been moving away from the tplNewPoint as they are all declared here. So, the tpl is redundent.

**Pull request merged**